frontend/src/js/store/devicesSlice/thunks.test.tsx

```diff
@@ -1185,7 +1186,40 @@ describe('troubleshooting related actions', () => {
 
     expect(result).toMatchObject({ start: new Date(endDate), end: new Date(endDate) });
   });
-
+  it('should allow triggering device inventory updates', async () => {
+    const store = mockStore({ ...defaultState });
+    const { attributes, id } = defaultState.devices.byId.a1;
+    const expectedActions = [
+      { type: triggerDeviceUpdate.pending.type },
+      { type: getDeviceById.pending.type },
+      { type: actions.receivedDevice.type, payload: { attributes, id } },
+      { type: getDeviceById.fulfilled.type },
+      { type: triggerDeviceUpdate.fulfilled.type }
+    ];
+    // no await here to allow moving beyond the delayed device info update in the next line
+    store.dispatch(triggerDeviceUpdate({ id: defaultState.devices.byId.a1.id, type: 'inventoryUpdate' }));
+    await jest.advanceTimersByTimeAsync(TIMEOUTS.fiveSeconds);
+    const storeActions = store.getActions();
+    expect(storeActions.length).toEqual(expectedActions.length);
+    expectedActions.map((action, index) => expect(storeActions[index]).toMatchObject(action));
```

Comment:
`forEach` here and line 1221 as well

Reply:
Good point! adjusted it in all the tests & the rest of the codebase where applicable.